reference/forms/types/money.rst

```diff
@@ -76,6 +76,23 @@ If set to ``true``, the HTML input will be rendered as a native HTML5
 
     As HTML5 number format is normalized, it is incompatible with ``grouping`` option.
 
+model_type
+~~~~~~~~~~
+
+**type**: ``string`` **default**: ``float``
+
+If, for some reason, you need the value to be converted to an ``integer`` instead of a ``float``,
+you can set the option value to ``integer``.
+
+.. caution::
+
+    When the value is set to ``integer``, the ``divisor`` option must not be set to ``1``,
+    to avoid losing decimals during the casting.
```

Comment:
The reason looks confusing to me. How is it possible to lose decimals when the divisor is `1`?

Looking at the implementation, it seems more related to the fact that the `integer` value doesn't have any effect when the divisor is `1`.

Reply:
I misspoke, in fact, it's more to do with the fact that it's an integer and a divisor of 1.

I hesitated for a long time before adding the caution message. Do you think it's still useful?

Reply:
@mdoutreluingne I still don't understand this caution. But it's not because of your PR. I've checked the code and I don't understand it either. I also read this discussion and I still don't understand it: https://github.com/symfony/symfony/pull/51884#discussion_r1399420685

How would you explain this in a way that anybody can understand it? Thanks.

Reply:
@javiereguiluz

From what I've understood from looking at the implementation, an exception is thrown if the `model_type` is `integer` and the divisor is `1`, because according to [this condition](https://github.com/garak/symfony/blob/881174157bc4a57863280d9033b416101725acb2/src/Symfony/Component/Form/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformer.php#L66) the casting is only done if the divisor is not equal to `1`, otherwise it only returns the value entered.

Me too, I don't understand why an exception is thrown if the divisor equals `1`. Casting can be done perfectly well if the divisor is equal to `1`.

I've added this caution message to warn the reader that if he changes the `model_type` for `integer`, he must also change the `divisor`, otherwise an exception will be thrown.

But maybe I've added a message that is ultimately of no interest to the reader and the message of the exception is enough. What do you think?

Reply:
Yes, I think it's better to not add this to the docs. Developes will get a explicty exception message if they do "wrong".

But, maybe we should revisit this or refactor the code to make it easier to understand. It seems that nobody truly understands this code 😐

Reply:
Removing caution message done.

I agree, I think it should be refactored to make it easier to understand.

Reply:
I've opened https://github.com/symfony/symfony/issues/53488 to see if we can finally understand the reason of this code. Thanks.